tsdb/executor.go

```diff
@@ -157,10 +157,18 @@ func (e *Executor) executeRaw(out chan *influxql.Row) {
 		}
 	}
 
-	// Get the union of SELECT fields across all mappers.
-	selectFields := newStringSet()
-	for _, m := range e.mappers {
-		selectFields.add(m.Fields()...)
+	// Get the distinct fields across all mappers.
+	var selectFields, aliasFields []string
+	if e.stmt.HasWildcard() {
+		sf := newStringSet()
```

Comment:
Should this still be a  stringset? What about `select mean(value), max(value) from foo`?